mantidproject/mantid
# Make SetGoniometer work on PeaksWorkspace et al.

**Description of work.**

This allows setting the Goniometer on a peaks workspace so that peaks can be created from the workspace with different goniometers.


**To test:**

```python
from mantid.simpleapi import *
from mantid.kernel import V3D

ws=CreateSampleWorkspace()
peaks=CreatePeaksWorkspace(ws, NumberOfPeaks=0)
peaks.addPeak(peaks.createPeak(V3D(1,0,1), 1))
SetGoniometer(peaks, Axis0='45,0,1,0,1')
peaks.addPeak(peaks.createPeak(V3D(1,0,1), 1))
SetGoniometer(peaks, Axis0='45,0,0,1,1')
peaks.addPeak(peaks.createPeak(V3D(1,0,1), 1))

for p in range(peaks.getNumberPeaks()):
    print("Peak {} =\n".format(p), peaks.getPeak(p).getGoniometerMatrix())
```

you will see that the peaks are created using the goniometer set on the workspace

```
Peak 0 =
 [[ 1.  0.  0.]
 [ 0.  1.  0.]
 [ 0.  0.  1.]]
Peak 1 =
 [[ 0.70710678  0.          0.70710678]
 [ 0.          1.          0.        ]
 [-0.70710678  0.          0.70710678]]
Peak 2 =
 [[ 0.70710678 -0.70710678  0.        ]
 [ 0.70710678  0.70710678  0.        ]
 [ 0.          0.          1.        ]]
```

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
